packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -0,0 +1,223 @@
+   */
+  message: PropTypes.string,
+  /**
+   * The Action of the primary button.
+   */
+  primaryAction: PropTypes.shape({
+    text: PropTypes.string,
+    onClick: PropTypes.func,
+  }),
+  /**
+   * The Action of the secondary button.
+   */
+  secondaryAction: PropTypes.shape({
+    text: PropTypes.string,
+    onClick: PropTypes.func,
+  }),
+  /**
+   * The variant of notification to be rendered.
+   * Use one of alert, error, warning, info, success, custom.
+   */
+  variant: PropTypes.oneOf([
+    variants.ALERT,
+    variants.ERROR,
+    variants.WARNING,
+    variants.INFO,
+    variants.SUCCESS,
+    variants.CUSTOM,
+  ]),
+  /**
+   * The icon to be used for a notification of type custom. This will not be used for any other notification types.
+   */
+  customIcon: PropTypes.element,
+  /**
+   * Toggle to show notification-dialog or not.
+   */
+  isOpen: PropTypes.bool.isRequired,
+  /**
+   * Toggle to focus notification-dialog or not.
+   */
+  isFocused: PropTypes.bool,
+  /**
+   * Callback function indicating a close condition was met, should be combined with isOpen for state management.
+   */
+  onRequestClose: PropTypes.func.isRequired,
+  /**
+   * A callback function to let the containing component (e.g. modal) to regain focus.
+   */
+  releaseFocus: PropTypes.func,
+  /**
+   * A callback function to request focus from the containing component (e.g. modal).
+   */
+  requestFocus: PropTypes.func,
+};
+
+const defaultProps = {
+  title: null,
+  message: null,
+  isFocused: false,
+  variant: variants.CUSTOM,
+};
+
+const actionSection = (primaryAction, secondaryAction) => {
+  let actionButton = null;
+  let dismissButton = null;
+  if (!primaryAction && !secondaryAction) {
+    return null;
+  }
+  if (primaryAction) {
+    actionButton = <Button text={primaryAction.text} variant={Button.Opts.Variants.EMPHASIS} onClick={primaryAction.onClick} />;
+  }
+  if (secondaryAction) {
+    dismissButton = <Button text={secondaryAction.text} onClick={secondaryAction.onClick} />;
+  }
+
+  return <div className={cx('actions')}>{actionButton}{dismissButton}</div>;
+};
+
+const getIcon = (variant, customIcon = null) => {
+  switch (variant) {
+    case variants.ALERT:
+      return (<IconAlert />);
+    case variants.ERROR:
+      return (<IconError />);
+    case variants.WARNING:
+      return (<IconWarning />);
+    case variants.INFO:
+      return (<IconInformation />);
+    case variants.SUCCESS:
+      return (<IconSuccess />);
+    case variants.CUSTOM:
+      return customIcon;
+    default:
+      return null;
+  }
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class NotificationDialog extends React.Component {
+
+  componentDidMount() {
+    if (this.props.isFocused && this.props.isOpen && this.props.requestFocus) {
```

Comment:
For the time being I think we only need the isFocused on the dialog modal version, since the notification dialog is the highest level component, its focus should always take focus.